internal/cmd/gocdk/internal/static/static.go

```diff
@@ -14,12 +14,285 @@
 
 package static
 
-import "net/http"
-
-// Open opens a file from the static assets.
-// TODO(rvangent): Provide a richer (but hopefully smaller) API,
-// which should cover things like copying files from assets/,
-// materializeTemplateDir, etc.
-func Open(path string) (http.File, error) {
-	return assets.Open(path)
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"text/template"
+
+	"golang.org/x/xerrors"
+)
+
+// List returns a listing of files and directories from the static data under
+// dir. If recurse is true, it recurses into directories.
+func List(dir string, recurse bool) ([]os.FileInfo, error) {
+	f, err := assets.Open(dir)
+	if err != nil {
+		return nil, err
+	}
+	infos, err := f.Readdir(-1)
+	if err != nil {
+		return nil, err
+	}
+	var retval []os.FileInfo
+	for _, info := range infos {
+		retval = append(retval, info)
+		if recurse && info.IsDir() {
+			r, err := List(path.Join(dir, info.Name()), true)
+			if err != nil {
+				return nil, err
+			}
+			retval = append(retval, r...)
+		}
+	}
+	return retval, nil
+}
+
+// MaterializeOptions holds options for Materialize.
+type MaterializeOptions struct {
+	// Force allows Materialize to clobber existing files when copying.
+	Force bool
+	// Logger is used to log information about changes.
+	Logger *log.Logger
+	// TemplateData is passed when executing templates marked with
+	// fileOptions.IsTemplate.
+	TemplateData interface{}
+}
+
+// TODO(rvangent): There are several cases where multiple places under
+// static/assets_ are repeating the same data:
+// 1. For places that we ask the user questions; currently, we are hardcoding
+//    their "answer" (e.g., in provision/blob/gcsblob/mainlocal_gcpproject.tf).
+//    But each place that wants to ask the question would need to repeat the
+//    .tf file and the .json config for it. Instead, we should have something
+//    like a "prompt" package that handles asking questions and (if needed)
+//    saving them. Each "provision" provider should just specify which
+//    questions it needs answered.
+// 2. Similarly, there are shared resources like the "random" provider that are
```

Comment:
I think this can be tackled by something like a `shared` directory in `static`, which the `.json` file can reference. E.g., `Path = ../shared/random_provider.tf`.